### PR TITLE
Metrics off

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### Reference Issues/PRs
#341 
#### What does this implement/fix? Explain your changes.
- `metrics_off` parameter added to ConfusionMatrix `__init__` method

### CODE REVIEW

File: pycm/pycm_param.py [hunk 1]
```diff
@@ -68,6 +68,7 @@
 VECTOR_ONLY_ERROR = "This option only works in vector mode"
 VECTOR_UNIQUE_CLASS_ERROR = "The classes list isn't unique. It contains duplicated labels."
 CLASS_NUMBER_ERROR = "Number of the classes is lower than 2"
+METRICS_OFF_ERROR = "This method doesn't work in `metrics off` mode."
```
Comment (sepandhaghighi) [inline pycm/pycm_param.py:71]: @alirezazolanvari 
Would you happen to have any suggestions here?

Comment (alirezazolanvari) [reply pycm/pycm_param.py:71]: My suggestion is:
"This method cannot be executed while `metrics_off = True`. "

Comment (sadrasabouri) [reply pycm/pycm_param.py:71]: `metrics_off=True`

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Minor comments have been made.

File: Test/plot_test.py [hunk 2]
```diff
@@ -244,4 +244,22 @@
...
+array([0, 1, 2])
+>>> ax.get_xticklabels()
+[Text(0, 0, '0'), Text(1, 0, '1'), Text(2, 0, '2')]
+>>> ax.get_yticks()
+array([0, 1, 2])
+>>> list(ax.get_yticklabels())
+[Text(0, 0, '0'), Text(0, 1, '1'), Text(0, 2, '2')]
+>>> list(ax.texts)
+[]
 """
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply Test/plot_test.py]: fixed in https://github.com/sepandhaghighi/pycm/pull/490/commits/5028d067c24e22e9ca542812e3ff90d37b25976f

File: Test/output_test.py [hunk 3]
```diff
@@ -561,6 +561,174 @@
...
+>>> save_stat=cm.save_html("test_metrics_off", address=False)
+>>> save_stat=={'Status': True, 'Message': None}
+True
+>>> save_stat=cm.save_csv("test_metrics_off", address=False)
+>>> save_stat=={'Status': True, 'Message': None}
+True
+>>> save_obj=cm.save_obj("test_metrics_off", address=False)
+>>> save_obj=={'Status': True, 'Message': None}
+True
+>>> cm_file_metrics_off=ConfusionMatrix(file=open("test_metrics_off.obj", "r"), metrics_off=True)
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply Test/output_test.py]: fixed in https://github.com/sepandhaghighi/pycm/pull/490/commits/5028d067c24e22e9ca542812e3ff90d37b25976f

File: Test/output_test.py [hunk 4]
```diff
@@ -561,6 +561,174 @@
...
+>>> save_stat=cm.save_stat("test_metrics_off", address=False)
+>>> save_stat=={'Status': True, 'Message': None}
+True
+>>> save_stat=cm.save_html("test_metrics_off", address=False)
+>>> save_stat=={'Status': True, 'Message': None}
+True
+>>> save_stat=cm.save_csv("test_metrics_off", address=False)
+>>> save_stat=={'Status': True, 'Message': None}
+True
+>>> save_obj=cm.save_obj("test_metrics_off", address=False)
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply Test/output_test.py]: fixed in https://github.com/sepandhaghighi/pycm/pull/490/commits/5028d067c24e22e9ca542812e3ff90d37b25976f

File: Test/output_test.py [hunk 5]
```diff
@@ -561,6 +561,174 @@
...
+>>> y_test = np.array([600, 200, 200, 200, 200, 200, 200, 200, 500, 500, 500, 200, 200, 200, 200, 200, 200, 200, 200, 200])
+>>> y_pred = np.array([100, 200, 200, 100, 100, 200, 200, 200, 100, 200, 500, 100, 100, 100, 100, 100, 100, 100, 500, 200])
+>>> cm=ConfusionMatrix(y_test, y_pred, metrics_off=True)
+>>> save_stat=cm.save_stat("test_metrics_off", address=False)
+>>> save_stat=={'Status': True, 'Message': None}
+True
+>>> save_stat=cm.save_html("test_metrics_off", address=False)
+>>> save_stat=={'Status': True, 'Message': None}
+True
+>>> save_stat=cm.save_csv("test_metrics_off", address=False)
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply Test/output_test.py]: fixed in https://github.com/sepandhaghighi/pycm/pull/490/commits/5028d067c24e22e9ca542812e3ff90d37b25976f

File: Test/output_test.py [hunk 6]
```diff
@@ -561,6 +561,174 @@
 >>> save_csv_data = cm.save_csv("test_header", header=True, matrix_save=True)
 >>> save_csv_data["Status"]
 True
+>>> y_test = np.array([600, 200, 200, 200, 200, 200, 200, 200, 500, 500, 500, 200, 200, 200, 200, 200, 200, 200, 200, 200])
+>>> y_pred = np.array([100, 200, 200, 100, 100, 200, 200, 200, 100, 200, 500, 100, 100, 100, 100, 100, 100, 100, 500, 200])
+>>> cm=ConfusionMatrix(y_test, y_pred, metrics_off=True)
+>>> save_stat=cm.save_stat("test_metrics_off", address=False)
+>>> save_stat=={'Status': True, 'Message': None}
+True
+>>> save_stat=cm.save_html("test_metrics_off", address=False)
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply Test/output_test.py]: fixed in https://github.com/sepandhaghighi/pycm/pull/490/commits/5028d067c24e22e9ca542812e3ff90d37b25976f

File: Test/output_test.py [hunk 7]
```diff
@@ -561,6 +561,174 @@
 >>> save_csv_data = cm.save_csv("test_header", header=True, matrix_save=True)
 >>> save_csv_data["Status"]
 True
+>>> y_test = np.array([600, 200, 200, 200, 200, 200, 200, 200, 500, 500, 500, 200, 200, 200, 200, 200, 200, 200, 200, 200])
+>>> y_pred = np.array([100, 200, 200, 100, 100, 200, 200, 200, 100, 200, 500, 100, 100, 100, 100, 100, 100, 100, 500, 200])
+>>> cm=ConfusionMatrix(y_test, y_pred, metrics_off=True)
+>>> save_stat=cm.save_stat("test_metrics_off", address=False)
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply Test/output_test.py]: fixed in https://github.com/sepandhaghighi/pycm/pull/490/commits/5028d067c24e22e9ca542812e3ff90d37b25976f

**Review by sadrasabouri — APPROVED** [review]

**Review by alirezazolanvari — APPROVED** [review]

**Pull request merged**